src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +80,11 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+
+  /** Returns the total number of messages for all users. */
+  public int getTotalMessageCount(){
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    return results.countEntities(FetchOptions.Builder.withLimit(1000));
```

Comment:
What is this magic number 1000?

Reply:
It is the default fetch limit when we call countEntities() function without parameter